packages/dashboard/src/pages/Personas/hooks/useOperateBindingProof.ts

```diff
@@ -0,0 +1,24 @@
+import type { ECKeyIdentifier } from '@masknet/shared-base'
+import { bindProof, createPersonaPayload } from '@masknet/web3-providers'
+import { useAsyncFn } from 'react-use'
+import { Services } from '../../../API'
+
+export function useOperateBound() {
+    return useAsyncFn(async (identifier: ECKeyIdentifier, profile, network, action) => {
+        const persona_ = await Services.Identity.queryPersona(identifier)
+        const username = profile.userId.toLowerCase()
+        const platform = network.split('.')[0] || 'twitter'
+        if (!persona_ || !persona_.publicHexKey) throw new Error('Failed to get person')
+        const payload = await createPersonaPayload(persona_.publicHexKey, action, username, platform)
```

Comment:
The proof server will give error message in response when create payload failed.